Example/Example/UI/Weather/WeatherViewController.swift

```diff
@@ -8,94 +8,109 @@
-        weatherModel.fetchWeather(at: "tokyo", date: Date()) { result in
-            DispatchQueue.main.async {
-                self.activityIndicator.stopAnimating()
-                self.handleWeather(result: result)
-            }
-        }
-        disasterModel.fetchDisaster { (disaster) in
-            self.disasterLabel.text = disaster
-        }
+    @IBAction func reload(_ sender: Any?) {
+        loadWeather()
     }
     
-    func handleWeather(result: Result<Response, WeatherError>) {
-        switch result {
-        case .success(let response):
-            self.weatherImageView.set(weather: response.weather)
-            self.minTempLabel.text = String(response.minTemp)
-            self.maxTempLabel.text = String(response.maxTemp)
-            
-        case .failure(let error):
-            let message: String
-            switch error {
-            case .jsonEncodeError:
-                message = "Jsonエンコードに失敗しました。"
-            case .jsonDecodeError:
-                message = "Jsonデコードに失敗しました。"
-            case .unknownError:
-                message = "エラーが発生しました。"
+    private func loadWeather() {
+        showIndicator { [weak self] in
+            self?.weatherModel.fetchWeather(at: "tokyo", date: Date()) { [weak self] result in
+                self?.mainQueueScheduler.schedule() {
+                    self?.handleWeather(result: result)
+                }
             }
             
-            let alertController = UIAlertController(title: "Error", message: message, preferredStyle: .alert)
-            alertController.addAction(UIAlertAction(title: "OK", style: .default) { _ in
-                self.dismiss(animated: true) {
-                    print("Close ViewController by \(alertController)")
+            self?.disasterModel.fetchDisaster { disaster in
+                self?.mainQueueScheduler.schedule() {
+                    self?.disasterLabel.text = disaster
                 }
-            })
-            self.present(alertController, animated: true, completion: nil)
+            }
+            
+        }
+    }
+    
+    private func showIndicator(while processings: () -> Void...) {
+        let dispatchGroup = DispatchGroup()
+        
+        switchView()
+        
+        processings.forEach { processing in
+            globalQueueScheduler.schedule(group: dispatchGroup) {
+                processing()
+            }
+        }
+        
+        dispatchGroup.notify(queue: .main) { [weak self] in
+            self?.switchView()
         }
     }
+    
+    private func switchView() {
+        activityIndicator.toggleAnimation()
+        reloadButton.toggleEnabled()
+        closeButton.toggleEnabled()
+    }
+    
+    private func handleWeather(result: Result<Response, WeatherError>) {
+        weatherHandler.handle(from: self, result)
+    }
+}
+
+
+private extension UIButton {
+    func toggleEnabled() {
+        self.isEnabled = !self.isEnabled
```

Comment:
[NITS] use toggle()
```suggestion
        self.isEnabled.toggle()
```
https://developer.apple.com/documentation/swift/bool/2994863-toggle